ui/src/pages/frontsite/Documents/DocumentsDetails/LoanRequestForm/LoanRequestForm.js

```diff
@@ -60,7 +68,9 @@ export default class LoanRequestForm extends Component {
     if (!isEmpty(this.state.deliveryMethod)) {
       optionalParams.deliveryMethod = this.state.deliveryMethod;
     }
-    this.props.requestLoanForDocument(documentPid, optionalParams);
+    this.props
+      .requestLoanForDocument(documentPid, optionalParams)
+      .then(() => this.handleErrorMessage());
```

Comment:
It would be better to use more `redux` here, rather than this technique.
The idea is that you perform an action (request loan) that has 3 possible states:
- loading
- success, with data
- error, with data
The state is updated in redux, and then injected via props.
When props are changed, React will re-render your component.

Basically, what I am saying is that you don't need this `then` and the `handleErrorMessage` function. We can discuss tomorrow by voice! :)